base/strings/string.jl

```diff
@@ -106,111 +211,172 @@ function first_utf8_byte(ch::Char)
 end
 
 function reverseind(s::String, i::Integer)
-    j = length(s.data) + 1 - i
-    d = s.data
-    while is_valid_continuation(d[j])
+    j = s.len + 1 - i
+    p = pointer(s)
+    while is_valid_continuation(unsafe_load(p,j))
```

Comment:
Would using `codeunit` be slower here? I would find it slightly nicer. (Same above.)